compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -625,8 +625,9 @@ impl<'tcx> Printer<'tcx> for SymbolMangler<'tcx> {
             ty::Uint(_) | ty::Int(_) | ty::Bool | ty::Char => {
                 ct_ty.print(self)?;
 
-                let mut bits = ct
-                    .try_to_bits(self.tcx, ty::TypingEnv::fully_monomorphized())
+                let (valtree, ty) = ct.to_valtree();
+                let mut bits = valtree
+                    .try_to_bits(self.tcx, ty, ty::TypingEnv::fully_monomorphized())
```

Comment:
This function already extracts the type and valtree, no need to do it again:

```suggestion
                let mut bits = valtree
                    .try_to_bits(self.tcx, ct_ty, ty::TypingEnv::fully_monomorphized())
```